poem/src/server.rs

```diff
@@ -93,8 +93,9 @@ where
         let Server { listener, name } = self;
         let name = name.as_deref();
         let alive_connections = Arc::new(AtomicUsize::new(0));
-        let notify = Arc::new(Notify::new());
-        let timeout_notify = Arc::new(Notify::new());
+        let cancellation_token = CancellationToken::new();
+        let timeout_token = CancellationToken::new();
+        let conn_shutdown_token = CancellationToken::new();
```

Comment:
I refactored it to `CancellationToken` instead of `Notify` because:
- `Notify::notify_waiters()` will notify only registered `Notified` (i.e `notify.registred()` is already called)
- We spawn tokio task to serve connection and register it inside (like `timeout_notify.notified()`), while spawned task can be polled with delay

So it's just more reliable structure here